### PR TITLE
PLAT-129757: Sync agate/Switch with Sandstone

### PR DESCRIPTION
**Pull request by daniel-stoian-lgp**
### Checklist

* [x] I have read and understand the [contribution guide](http://enactjs.com/docs/developer-guide/contributing/)
* [ ] A CHANGELOG entry is included
* [x] At least one test case is included for this feature or bug fix
* [x] Documentation was added or is not needed
* [ ] This is an API breaking change

### Issue Resolved / Feature Added
[//]: # (Describe the issue resolved or feature added by this pull request)
Feature added: agate/Switch:
- added `disabled` and selected `props`
- changed component structure
- deprecated internal/ToggleIcon 

### Resolution
[//]: # (Does the code work as intended?)
[//]: # (What is the impact of this change and *why* was it made?)
The change was needed in order to synchronize the component with Sandstone

### Links
[//]: # (Related issues, references)
PLAT-129757

### Comments

Enact-DCO-1.0-Signed-off-by: Daniel Stoian daniel.stoian@lgepartner.com

### CODE REVIEW

**Review by stanca-pop-lgp — APPROVED** [review]
LGTM

**Review by adrian-cocoara-lgp — APPROVED** [review]

File: Switch/Switch.js [hunk 1]
```diff
@@ -31,9 +35,40 @@ const SwitchBase = kind({
...
+		 *
+		 * May be specified as either:
+		 *
+		 * * A string that represents an icon from the [iconList]{@link agate/Icon.Icon.iconList},
+		 * * An HTML entity string, Unicode reference or hex value (in the form '0x...'),
+		 * * A URL specifying path to an icon image, or
+		 * * An object representing a resolution independent resource (See {@link ui/resolution})
+		 *
+		 * @see {@link agate/Icon.IconBase.children}
+		 * @type {String|Object}
```
Comment (MikyungKim) [inline Switch/Switch.js:49]:
```suggestion
		 * @type {String}
```

File: Switch/Switch.js [hunk 2]
```diff
@@ -31,9 +35,40 @@ const SwitchBase = kind({
...
+		 * @public
+		 */
 		children: PropTypes.string,
```
Comment (MikyungKim) [inline Switch/Switch.js:52]:
```suggestion
		children: PropTypes.oneOfType([PropTypes.string, PropTypes.object]),
```

File: Switch/Switch.js [hunk 3]
```diff
@@ -31,9 +35,40 @@ const SwitchBase = kind({
...
+		 * @type {Object}
+		 * @public
+		 */
 		css: PropTypes.object,
 
+		/**
+		 * Disables Switch and becomes non-interactive.
+		 *
+		 * @type {Boolean}
+		 * @default false
```
Comment (MikyungKim) [inline Switch/Switch.js:67]:
```suggestion
```

File: Switch/Switch.js [hunk 4]
```diff
@@ -42,6 +77,22 @@ const SwitchBase = kind({
 		 * @public
 		 */
 		noAnimation: PropTypes.bool,
+
+		/**
+		 * Sets whether this control is in the 'on' or 'off' state. `true` for 'on', `false` for 'off'.
+		 *
+		 * @type {Boolean}
+		 * @default false
```
Comment (MikyungKim) [inline Switch/Switch.js:85]:
```suggestion

```

File: CHANGELOG.md [hunk 5]
```diff
@@ -17,6 +17,8 @@ The following is a curated list of changes in the Enact agate module, newest cha
 - `agate/Popup` content padding and margin with `closeButton`
 - `agate/Popup` `bottom`, `fullscreen`, `left`, and `right` values to `position` prop
 - `agate/Popup` `duration` and `type` props
+- `agate/Switch` prop `disabled` to disable switch and become non-interactive
+- `agate/Switch` prop `selected` to set whether the control is in the 'on' or 'off' state
```
Comment (MikyungKim) [inline CHANGELOG.md:21]: Please remove the changelog since `agate/Switch` is a private component.

File: Switch/Switch.js [hunk 6]
```diff
@@ -42,6 +77,22 @@ const SwitchBase = kind({
...
+		 * @public
+		 */
+		selected: PropTypes.bool,
+
+		/**
+		 * The current skin for this component.
+		 *
+		 * @type {String}
+		 * @public
```
Comment (MikyungKim) [inline Switch/Switch.js:94]:
```suggestion
		 * @private
```

File: Switch/Switch.js [hunk 7]
```diff
@@ -67,21 +120,62 @@ const SwitchBase = kind({
 		}
 	},
 
-	render: ({css, ...rest}) => {
+	render: ({children, disabled, css, selected, ...rest}) => {
```
Comment (MikyungKim) [inline Switch/Switch.js:123]:
```suggestion
	render: ({children, css, disabled, selected, ...rest}) => {
```

File: Switch/Switch.js [hunk 8]
```diff
@@ -31,9 +35,40 @@ const SwitchBase = kind({
 	name: 'Switch',
```
Comment (MikyungKim) [inline Switch/Switch.js:35]: At line 25, the class name should be `SwitchBase`.

**Review by MikyungKim — CHANGES_REQUESTED** [review]
Thanks! Please find my comments.

File: CHANGELOG.md (hunk 5 above)
Comment (daniel-stoian-lgp) [reply CHANGELOG.md:21]: Thanks. Good to know. I will share with the others as well

Comment (MikyungKim) [reply CHANGELOG.md]: Great! 👍

**Review by MikyungKim — APPROVED** [review]
LGTM

**Pull request merged**